### PR TITLE
chore: Python 3.9

### PR DESCRIPTION
**Pull request by frankh**
## Problem

Newer python = better python

## Changes

👉 *Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review.*

## How did you test this code?

- [x] CI passes
- [x] runs locally

### CODE REVIEW

**Review by hazzadous — APPROVED** [review]
Tests pass and don't see anything big by way of other changes.

Is it just pytest version bump and the xmlsec stuff? (plus unrelated snapshot updates)

**Comment by hazzadous** [comment]
I'll probably need to update my codespaces config as well.

File: posthog/api/test/__snapshots__/test_decide.ambr [hunk 1]
```diff
@@ -589,7 +589,9 @@
 ---
 # name: TestDecide.test_flag_with_regular_cohorts.5
   '
-  SELECT ("posthog_person"."properties" -> '$some_prop_1') = '"something_1"' AS "flag_X_condition_0"
+  SELECT (("posthog_person"."properties" -> '$some_prop_1') = '"something_1"'
```
Comment (hazzadous) [inline posthog/api/test/__snapshots__/test_decide.ambr:592]: @neilkakkar do you think these changes are safe and/or unrelated?

**Comment by frankh** [comment]
uhh apparently I hit the merge button? 😬 

I'm sure it will be fine

**Comment by hazzadous** [comment]
@frankh I'll keep an eye

File: posthog/api/test/__snapshots__/test_decide.ambr (hunk 1 above)
Comment (neilkakkar) [reply posthog/api/test/__snapshots__/test_decide.ambr:592]: Safe! How are they still showing up on PRs when I merged the fix -.-

**Pull request merged**